include/cjose/jwe.h

```diff
@@ -42,21 +44,45 @@ typedef struct _cjose_jwe_int cjose_jwe_t;
  * (e.g. (dir), the provided JWK must be symmetric (e.g. oct).
  *
  * \param jwk [in] the key to use for encrypting the JWE.
- * \param protected_header [in] additional header values to include in the JWE header.
+ * \param protected_header [in] additional header values to include in the JWE protected header.
  * \param plaintext [in] the plaintext to be encrypted in the JWE payload.
  * \param plaintext_len [in] the length of the plaintext.
  * \param err [out] An optional error object which can be used to get additional
  *        information in the event of an error.
  * \returns a newly generated JWE with the given plaintext as the payload.
  */
-cjose_jwe_t *cjose_jwe_encrypt(
-    const cjose_jwk_t *jwk, cjose_header_t *protected_header, const uint8_t *plaintext, size_t plaintext_len, cjose_err *err);
+cjose_jwe_t *
+cjose_jwe_encrypt(const cjose_jwk_t *jwk, cjose_header_t *header, const uint8_t *plaintext, size_t plaintext_len, cjose_err *err);
 
 /**
- * Creates a serialization of the given JWE object.
- *
- * Note the current implementation only supports serialization to the JWE
- * compact format.
+ * Creates a new JWE by encrypting the given plaintext with multiple keys.
+ * \see ::cjose_jwe_encrypt for key requirements.
+ * \param jwk [in] array of keys to use for encrypting the JWE.
+ * \param unprotected_header [in] array of unprotected headers, one matching each key entry.
+ *        An element can be NULL if no additional header fields are needed. The headers
+ *        are retained by the JWE and can be released by the caller if no longer needed.
+ * \param jwk_len [in] number of entries in both `jwk` and `unprotected_header` arrays.
+ * \param protected_header [in] additional header values to include in the JWE protected header. The header
+ *        is retained by JWE and should be released by the caller if no longer needed.
+ * \param unprotected_header [in] additional header values to include in the shared JWE unprotected header,
+ *        can be NULL. The header is retained by JWE and should be released by the caller if no longer needed.
+ * \param plaintext [in] the plaintext to be encrypted in the JWE payload.
+ * \param plaintext_len [in] the length of the plaintext.
+ * \param err [out] An optional error object which can be used to get additional
+ *        information in the event of an error.
+ * \returns a newly generated JWE with the given plaintext as the payload.
+ */
+cjose_jwe_t *cjose_jwe_encrypt_full(const cjose_jwk_t **jwk,
```

Comment:
Instead of two separate arrays that have to match up, what about making a public `cjose_jwe_recipient_t` struct that has the `jwk` and `unprotected_header`?